### PR TITLE
Add missing topics to namespaced consumer groups

### PR DESCRIPTION
**Pull request by yaron2**
Fixes https://github.com/dapr/dapr/issues/5472.

### CODE REVIEW

File: pkg/runtime/runtime.go [hunk 1]
```diff
@@ -1873,6 +1886,11 @@ func (a *DaprRuntime) Publish(req *pubsub.PublishRequest) error {
 	}
 
 	policy := a.resiliency.ComponentOutboundPolicy(a.ctx, req.PubsubName, resiliency.Pubsub)
+
+	if a.pubSubs[req.PubsubName].namespaceScoped {
```
Comment (Taction) [inline pkg/runtime/runtime.go:1890]: Please use `if ps.namespaceScoped {`, this map's read is under RLock.

Comment (yaron2) [reply pkg/runtime/runtime.go]: Done

**Comment by Taction** [comment]
Generally looks good to me, please check if `bulkSubscribeTopic` needs to remove the namespace prefix when sending messages to app. Also, it is best to have e2e tests to ensure these namespaced subscribe behaviors.

**Comment by yaron2** [comment]
> Generally looks good to me, please check if `bulkSubscribeTopic` needs to remove the namespace prefix when sending messages to app. Also, it is best to have e2e tests to ensure these namespaced subscribe behaviors.

Thanks, and you're right, this feature was enabled before bulk pub/sub so `bulkSubscribeTopic` needed an update, which I've now added. an e2 test for both namespaced publishers and subscribers is being worked on separately.

**Comment by Taction** [comment]
Another thing needs to be considered: does the dead-letter topic need to add a namespace prefix?

**Comment by yaron2** [comment]
> Another thing needs to be considered: does the dead-letter topic need to add a namespace prefix?

I see namespaced DLTs as almost a separate sub feature. It's optional, and I'm leaning towards adding support for it, but shouldn't be part of this PR.

**Review by Taction — APPROVED** [review]
Thanks for this pr and the explanation, LGTM.

**Pull request merged**